src/dataset.h

```diff
@@ -103,6 +103,13 @@ class Dataset {
   }
 
   void insert(Variable variable);
+  template <class T> void insert(const std::string &name, const T &slice) {
+    for (const auto &var : slice) {
+      Variable newVar(var);
+      newVar.setName(name);
+      this->insert(newVar);
+    }
```

Comment:
Doesn't this go wrong if the subset contains coordinates? Coordinates should probably be compared, without renaming them.